libnmstate/metadata.py

```diff
@@ -147,3 +153,22 @@ def _generate_link_master_metadata(ifaces_desired_state,
                         master_has_no_slaves_specified_in_desired):
                     set_metadata_func(
                         master_state, ifaces_desired_state[slave])
+
+
+def _generate_route_metadata(desired_state):
+    """
+    Save routes under interface for creating setting profile.
+    Currently route['next-hop-interface'] is mandatory.
```

Comment:
The comments here a not completely clear. After a lot of thinking I understood them but maybe you can rephrase them. It is irritating that the comment mentions the next hop interface but it is not explicitly used in the code AFAIU it is used by the iface_routes property. One guess to make it cleaner would be to have InterfaceState objects that handle the full state and properly generate them from the dict when the actual state object is created.

On second thought, maybe this function could be a method of the State class that is called in the constructor instead.

Reply:
Comment changed to:

```
Save routes under interface IP protocol so that nm/ipv4.py or nm/ipv6.py
could include route configuration in `create_setting()`.
Currently route['next-hop-interface'] is mandatory.
```

`InterfaceState` is too big for this PR. If possible, we need a new PR for the class/object
represented states.